core/tests/protractor/learnerFlow.js

```diff
@@ -0,0 +1,455 @@
+var LibraryPage = require('../protractor_utils/LibraryPage.js');
+
+describe('Learner dashboard functionality', function() {
+  var adminPage = null;
+  var creatorDashboardPage = null;
+  var explorationEditorPage = null;
+  var explorationEditorMainTab = null;
+  var explorationPlayerPage = null;
+  var libraryPage = null;
+  var learnerDashboardPage = null;
+  var oppiaLogo = element(by.css('.protractor-test-oppia-main-logo'));
+  var continueButton = element(by.css('.protractor-test-continue-button'));
+  var clickContinueButton = function() {
+    waitFor.elementToBeClickable(
+      continueButton, 'Could not click continue button');
+    continueButton.click();
+    waitFor.pageToFullyLoad();
+  };
+
+  var createDummyExplorationOnDesktop = function() {
+    creatorDashboardPage.get();
+    creatorDashboardPage.clickCreateActivityButton();
+    waitFor.pageToFullyLoad();
+    explorationEditorMainTab.exitTutorial();
+    explorationEditorMainTab.setStateName('First');
+    explorationEditorMainTab.setContent(forms.toRichText(
+      'Hi there, I’m Oppia! I’m an online personal tutor for everybody!'));
+    explorationEditorMainTab.setInteraction('Continue');
+    var responseEditor = explorationEditorMainTab.getResponseEditor('default');
+    responseEditor.setDestination('Second', true, null);
+    explorationEditorMainTab.moveToState('Second');
+    explorationEditorMainTab.setContent(forms.toRichText(
+      'So what can I tell you?'));
+    explorationEditorMainTab.setInteraction('MultipleChoiceInput', [
+      forms.toRichText('How do your explorations work?'),
+      forms.toRichText('What can you tell me about this website?'),
+      forms.toRichText('How can I contribute to Oppia?'),
+      forms.toRichText('Those were all the questions I had!')
+    ]);
+    explorationEditorMainTab.addResponse(
+      'MultipleChoiceInput', null, 'End Card', true, 'Equals',
+      'Those were all the questions I had!');
+    responseEditor = explorationEditorMainTab.getResponseEditor('default');
+    responseEditor.setFeedback(forms.toRichText('I do not know!'));
+    explorationEditorMainTab.moveToState('End Card');
+    explorationEditorMainTab.setContent(
+      forms.toRichText('Congratulations, you have finished!'));
+    explorationEditorMainTab.setInteraction('EndExploration');
+    explorationEditorPage.navigateToSettingsTab();
+    explorationEditorSettingsTab.setTitle('Dummy Exploration');
+    explorationEditorSettingsTab.setCategory('Algorithm');
+    explorationEditorSettingsTab.setObjective('Learn more about Oppia');
+    explorationEditorSettingsTab.setLanguage('English');
+    explorationEditorPage.saveChanges();
+    workflow.publishExploration();
+  };
+
+  beforeAll(function() {
+    adminPage = new AdminPage.AdminPage();
+    libraryPage = new LibraryPage.LibraryPage();
+    learnerDashboardPage = new LearnerDashboardPage.LearnerDashboardPage();
+
+    if (browser.isMobile) {
```

Comment:
In all such if/else statements, both branches should have the same functionality. This isn't the case here.

It might be better to do the branching on a per-test basis.

Reply:
Done, followed the per-test basis. Thanks!